pyomo/solvers/plugins/solvers/scip_direct.py

```
@@ -0,0 +1,838 @@
+            repn = generate_standard_repn(expr, quadratic=True)
+        else:
+            repn = generate_standard_repn(expr, quadratic=False)
+
+        scip_expr, referenced_vars = self._get_expr_from_pyomo_repn(repn, max_degree)
+
+        return scip_expr, referenced_vars
+
+    def _scip_lb_ub_from_var(self, var):
+        if var.is_fixed():
+            val = var.value
+            return val, val
+        if var.has_lb():
+            lb = value(var.lb)
+        else:
+            lb = -self._solver_model.infinity()
+        if var.has_ub():
+            ub = value(var.ub)
+        else:
+            ub = self._solver_model.infinity()
+        return lb, ub
+
+    def _add_var(self, var):
+        varname = self._symbol_map.getSymbol(var, self._labeler)
+        vtype = self._scip_vtype_from_var(var)
+        lb, ub = self._scip_lb_ub_from_var(var)
+
+        scip_var = self._solver_model.addVar(lb=lb, ub=ub, vtype=vtype, name=varname)
+
+        self._pyomo_var_to_solver_var_expr_map[var] = scip_var
+        self._pyomo_var_to_solver_var_map[var] = scip_var.name
+        self._solver_var_to_pyomo_var_map[varname] = var
+        self._referenced_variables[var] = 0
+
+    def close(self):
+        """Frees SCIP resources used by this solver instance."""
+
+        if self._solver_model is not None:
+            self._solver_model.freeProb()
+            self._solver_model = None
+
+    def __exit__(self, t, v, traceback):
+        super().__exit__(t, v, traceback)
+        self.close()
+
+    def _set_instance(self, model, kwds={}):
+        DirectOrPersistentSolver._set_instance(self, model, kwds)
+        try:
+            self._solver_model = self._scip.Model()
+        except Exception:
+            e = sys.exc_info()[1]
+            msg = (
+                "Unable to create SCIP model. "
+                "Have you installed PySCIPOpt correctly?\n\n\t"
+                + "Error message: {0}".format(e)
+            )
```

It may be preferable to check `self.available` first. If it's not available, then you can raise an error that it couldn't be found. If it's any _other_ error, you can say, "Hey, it's available, but something isn't working correctly. Here is your error message."

---

I added a call to `self.available()` as suggested. It can't hurt to be there. The call itself raises an appropriate warning by default if the python binding can't be found.